src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs

```diff
@@ -0,0 +1,257 @@
+            //Assert.Equal ((int)MonoErrorCodes.BpNotFound, bp1_res.Error ["code"]?.Value<int> ());
+        }
+
+        [Fact]
+        public async Task CreateGoodBreakpointAndHit()
+        {
+            var bp = await SetBreakpoint("dotnet://debugger-test.dll/debugger-test.cs", 10, 8);
+
+            var eval_req = JObject.FromObject(new
+            {
+                expression = "window.setTimeout(function() { invoke_add(); }, 1);",
+            });
+
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_add(); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs", 10, 8,
+                "IntAdd",
+                wait_for_event_fn: (pause_location) =>
+                {
+                    Assert.Equal("other", pause_location["reason"]?.Value<string>());
+                    Assert.Equal(bp.Value["breakpointId"]?.ToString(), pause_location["hitBreakpoints"]?[0]?.Value<string>());
+
+                    var top_frame = pause_location["callFrames"][0];
+                    Assert.Equal("IntAdd", top_frame["functionName"].Value<string>());
+                    Assert.Contains("debugger-test.cs", top_frame["url"].Value<string>());
+
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 8, 4, scripts, top_frame["functionLocation"]);
+
+                    //now check the scope
+                    var scope = top_frame["scopeChain"][0];
+                    Assert.Equal("local", scope["type"]);
+                    Assert.Equal("IntAdd", scope["name"]);
+
+                    Assert.Equal("object", scope["object"]["type"]);
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 8, 4, scripts, scope["startLocation"]);
+                    CheckLocation("dotnet://debugger-test.dll/debugger-test.cs", 14, 4, scripts, scope["endLocation"]);
+                    return Task.CompletedTask;
+                }
+            );
+        }
+
+        public static TheoryData<string, string, string, bool> FalseConditions = new TheoryData<string, string, string, bool>
+        {
+            { "invoke_add()", "IntAdd", "0.0", false },
+            { "invoke_add()", "IntAdd", "c == 40", false },
+            { "invoke_add()", "IntAdd", "c < 0", false },
+        };
+
+        public static TheoryData<string, string, string, bool> TrueConditions = new TheoryData<string, string, string, bool>
+        {
+            { "invoke_add()", "IntAdd", "c == 30", true },
+            { "invoke_add()", "IntAdd", "true", true },
+            { "invoke_add()", "IntAdd", "5", true },
+            { "invoke_add()", "IntAdd", "c < 40", true },
+            { "invoke_use_complex()", "UseComplex", "complex.A == 10", true },
+            { "invoke_add()", "IntAdd", "1.0", true },
+            { "invoke_add()", "IntAdd", "\"foo\"", true },
+            { "invoke_add()", "IntAdd", "\"true\"", true },
+            { "invoke_add()", "IntAdd", "\"false\"", true },
```

Comment:
Evaluating the condition which returns a string, or number, as `true`, and thus stopping at the breakpoint seems like a JS coercion like behavior. In chrome, it does stop for those. But in VS we get:

### `condition="4"`:

![Screen Shot 2021-02-27 at 7 10 13 PM](https://user-images.githubusercontent.com/1472/109403981-a397f200-792f-11eb-9a8d-fe2393dfe281.png)

### `condition="\"Asd\""`
![Screen Shot 2021-02-27 at 7 09 35 PM](https://user-images.githubusercontent.com/1472/109403982-a397f200-792f-11eb-9dfb-b0cfb774eac1.png)

I think we should have the same behavior for managed code. Evaluate to true only if the c# expression evaluates to a bool? Feel free to correct me though.

Also, on a related point, VS seems to throw up an error if the condition doesn't evaluate to a boolean. Should we be surfacing the error too? I'm not sure how we can get it to show up as a dialog like in the screenshots, but maybe we can log an error? Thoughts?

Maybe @lewing, or @captainsafia have some an opinion on this?

Reply:
Okay, when they comment and we decide I update my PR.

Reply:
I don't think there is an effective way to communicate this in the js interfaces right?  The breakpoint can be defined before we even know where it binds and there isn't a good way to communicate this when it is hit?

Reply:
Yes, I agree... there isn't any effective way to communicate it.

Reply:
Should we log that as an error then?

Reply:
Log as an error everything that does not returns a bool and don't stop, is this the behavior that we want?